apps/prairielearn/src/ee/pages/lti13CourseNavigation/lti13CourseNavigation.ts

```diff
@@ -36,6 +36,25 @@ function prettyCourseName(ltiClaim) {
   }
 }
 
+router.get(
+  '/course_instances',
+  asyncHandler(async (req, res) => {
+    const course_instances = await selectCourseInstancesWithStaffAccess({
```

Comment:
This `selectCourseInstancesWithStaffAccess()` function only looks for view access (`role > None`) and doesn't include any other information about access to filter after the fact.

This is probably what leads to confusion with course instance permissions that Matt started to document in #11055   If we did want to restrict LTI linking to course instance edit permissions, we may want to revisit this function or write a similar one to it for more granular access decisions.

Reply:
This PR addressed this by adding `has_course_instance_permission_edit` to the `selectCourseInstancesWithStaffAccess()` returned objects, and checking it.